app.py

```diff
@@ -3,38 +3,49 @@
 """
 
 import os
-import json
 import flask
-from flask import render_template, url_for, redirect, flash
-from flask_login import (
-    login_user,
-    login_required,
-    logout_user,
-)
-from setup import app, bcrypt
+from flask import flash
+from flask_login import current_user, login_user, login_required, logout_user
+from setup import app, bcrypt, csrf
 from database import db, BTUser, Profile
 from forms import LoginForm, RegisterForm, ProfileForm
 from fdc import search
 
 
-@app.route("/", methods=["GET", "POST"])
-@login_required
+def render(**kwargs):
+    """
+    Renders the page that handles routing on the frontend.
+    """
+
+    return flask.render_template("index.html", **kwargs)
+
+
+def redirect(route):
+    """
+    Handles redirection to a route.
+    """
+    return flask.redirect(flask.url_for(route))
+
+
+@app.route("/")
 def main():
     """
     The main landing page that is shown when the app is launced.
     """
-    return flask.render_template("index.html")
+    if current_user.is_authenticated:  # avoid "must login" message
+        return redirect("index")
+    return redirect("login")
 
 
-@app.route("/index", methods=["GET", "POST"])
+@app.route("/app")
```

Comment:
Will these not have methods?

Reply:
The GET method is the default if the method is not specified, and we aren't using the POST method on that endpoint (the frontend uses `fetch` with the `/api/search` endpoint).

Reply:
👍